text/0631-refresh-method-for-router-service.md

```diff
@@ -7,46 +7,63 @@
 
 ## Summary
 
-> Add a refresh method to the router service that calls refresh on all currently active routes.
+Add a refresh method to the router service that calls refresh on all currently active routes,
+or refreshes the descendents of the active route referred to by the pivot route name provided as an argument.
 
 ## Motivation
 
-> We want to be able to call refresh on all currently active routes from a centralized service
-or from a component. As a side benefit, we will be able to do this without relying on the send api,
+We want to be able to call refresh on all currently active routes, or a subset of them,
+from a centralized service or from a component.
+As a side benefit, we will be able to do this without relying on the send api,
 which is being discussed as a possible deprecation in RFC 632.
 This enables us to get the latest data from the model hook.
 
 ## Detailed design
 
+The following pseudocode represents what the overall technical design of this method.
+This code implementation is not normative.
+
 ```js
 class RouterService {
-    refresh() {
-        this._router._routerMicroLib.refresh();
+    refresh(pivotRouteName?: string): Promise<Transition> {
```

Comment:
As it stands today we should be returning a `Transition` which _is_ a "thennable" (not a promise that resolves to a transition).

```suggestion
    refresh(pivotRouteName?: string): Transition {
```